caching-service/src/main/java/org/zowe/apiml/caching/service/vsam/VsamFile.java

```diff
@@ -196,11 +196,22 @@ public void close() {
     }
 
     public List<VsamRecord> readForService(String serviceId) {
+        List<VsamRecord> returned = new ArrayList<>();
+
+        serviceWideOperation(serviceId, (zfile, record) -> returned.add(record));
+
+        return returned;
+    }
+
+    public void deleteForService(String serviceId) {
+        serviceWideOperation(serviceId, (zfile, record) -> zfile.delrec());
+    }
+
+    private void serviceWideOperation(String serviceId, ServiceWideOperation operation) {
```

Comment:
Method's purpose has changed, the debug logs could change too, to better reflect the new purpose

Reply:
Do you have any idea what exact logs? The ones inside looks like they could be well applied to both scenarios. Do you mean add log to the operation itself to distinguish?

Reply:
I have added explanation to the logs.